### ISSUE TITLE
implement "--log_file" and "--log_dir" for klog

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#9521]
I hardcoded this in our code to force log file to be to a specific file, 

```
	if err := goflag.Set("log_file", "medya_1.txt"); err != nil {
```

and makes the log file to append to medya_1.txt  we just need to parse this from the flags

however if I pass the same flag to the minikube binary it will not do anything (without that line above)

```
minikube status --log_file=medya2.txt
```

same with one dash

```
minikube status -log_file=medya2.txt
```
```

**Comment by medyagh** [comment]
I see we had a workaround package that we created to fix Glog erroring

```
	// initflag must be imported before any other minikube pkg.
	// Fix for https://github.com/kubernetes/minikube/issues/4866

	"k8s.io/klog/v2"
	_ "k8s.io/minikube/pkg/initflag"
https://github.com/medyagh/minikube/blob/717b738d6d785b21604f9145dc3432680034900a/pkg/minikube/extract/extract.go#L34
```


but I see after the klog refactor we are importing klog before initflag package (that could be an issue maybe?)

**Comment by medyagh** [comment]
I just did a standalone example Cobra+Klog and I noticed running this code does NOT produce any temp file logs AT ALL !!!
```
find $TMPDIR -mtime -1 -type f -name "*INFO*" -ls 2>/dev/null
```

that makes me think, is the temp file really created by klog ???

here is the code for the standalone example , same as (this llink)[https://flowerinthenight.com/blog/2019/02/05/golang-cobra-klog]  except I made it v2 to match minikube
```
package main

import (
	goflag "flag"

	"github.com/spf13/cobra"
	flag "github.com/spf13/pflag"
	"k8s.io/klog/v2"
)

var (
	str = "hello world"

	rootCmd = &cobra.Command{
		Use:   "echo",
		Short: "use klog with cobra",
		Long:  "Use klog together with cobra.",
	}
)

func init() {
	rootCmd.Flags().SortFlags = false
	rootCmd.AddCommand(
		RunCmd(),
	)

	klog.InitFlags(nil)
	goflag.Parse()
	flag.CommandLine.AddGoFlagSet(goflag.CommandLine)
}

func RunCmd() *cobra.Command {
	runcmd := &cobra.Command{
		Use:   "run",
		Short: "run command",
		Long:  "Run command.",
		Run: func(cmd *cobra.Command, args []string) {
			klog.Infof("echo=%v", str)
		},
	}

	runcmd.Flags().SortFlags = false
	runcmd.Flags().StringVar(&str, "str", str, "string to print")
	return runcmd
}

func main() {
	if err := rootCmd.Execute(); err != nil {
		klog.Fatalf("root cmd execute failed, err=%v", err)
	}
}


```

**Comment by medyagh** [comment]
@prezha thanks for taking this, please let me know if you go stuck. and please hit me up on slack !

btw I woud like to invite you to minikube office hours this monday ! we have Minikube Crazy Idea event. (to come up with crazy ideas for the next 2 years)

**Comment by prezha** [comment]
@medyagh no probs, i will

also, thank you for inviting me: i'd love to join minikube office hours on monday to the minikube crazy idea event - sounds like fun ;)

**Comment by medyagh** [comment]
I belive this must be because we use kubectls'  templates to group the cobra command as seen [here](https://github.com/medyagh/minikube/blob/c798567b729ea6aceedc09449f0ed0b0564d991b/cmd/minikube/cmd/root.go#L208)

 we might need to do what openshift is donig [here](https://github.com/ElwahidG/openshift/blob/97de2a03ef345405c311daa69d4b93ecdbd1e3d2/pkg/cmd/flagtypes/glog.go)

**Comment by prezha** [comment]
thank you @medyagh, i'll have a look at that

**Comment by prezha** [comment]
@medyagh i think that the issue is with go flag and the fact that we are using subcommands (that prevent go flag from parsing args); that can be solved with pflag - details are in the pr #9592 
please have a look and share your thoughts
